Core/Core/Api/GraphQL/Resources/CommentResource.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using GraphQL;
+using Speckle.Core.Api.GraphQL.Inputs;
+using Speckle.Core.Api.GraphQL.Models;
+using Speckle.Core.Api.GraphQL.Models.Responses;
+
+namespace Speckle.Core.Api.GraphQL.Resources;
+
+public sealed class CommentResource
+{
+  private readonly ISpeckleGraphQLClient _client;
+
+  internal CommentResource(ISpeckleGraphQLClient client)
+  {
+    _client = client;
+  }
+
+  /// <param name="projectId"></param>
+  /// <param name="limit">Max number of comments to fetch</param>
+  /// <param name="cursor">Optional cursor for pagination</param>
+  /// <param name="filter">Optional filter</param>
+  /// <param name="repliesLimit">Max number of comment replies to fetch</param>
+  /// <param name="repliesCursor">Optional cursor for pagination</param>
+  /// <param name="cancellationToken"></param>
+  /// <returns></returns>
+  /// <inheritdoc cref="ISpeckleGraphQLClient.ExecuteGraphQLRequest{T}"/>
+  public async Task<ResourceCollection<Comment>> GetProjectComments(
+    string projectId,
+    int limit = 25,
```

Comment:
would it be useful to make these constants and collate these in a sensible class or are they too disparate?